compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -999,6 +1002,45 @@ public void visit(BLangSimpleVariable varNode) {
         }
     }
 
+    private void analyzeArrayVariableImplicitInitialValue(BLangSimpleVariable varNode) {
+        if (varNode.typeNode == null || varNode.typeNode.type == null) {
+            return;
+        }
+        // Variable is a array def, elements must have implicit initial value.
+        if (varNode.typeNode.type.tag == TypeTags.ARRAY) {
+            analyzeArrayElemImplicitInitialValue(varNode.typeNode, varNode.pos);
+        } else if (varNode.typeNode.type.tag == TypeTags.UNION
+                && varNode.typeNode.getKind() == NodeKind.UNION_TYPE_NODE) {
+            // Check each member of the union.
+            for (BLangType memberTypeNode : ((BLangUnionTypeNode) varNode.typeNode).memberTypeNodes) {
+                analyzeArrayElemImplicitInitialValue(memberTypeNode, memberTypeNode.pos);
+            }
+        } else if (varNode.typeNode.type.tag == TypeTags.UNION && varNode.typeNode.getKind() == NodeKind.ARRAY_TYPE) {
```

Comment:
If we check `varNode.typeNode.getKind()` only for all three cases, then i think we can merge this one with L#1010.